### PR TITLE
Pinned Apps (1 of 2)

### PR DESCRIPTION
**Pull request by calebjacob**
Partially Implements: https://github.com/near/near-discovery/issues/1056

This PR should be merged before the 2nd PR to make testing easier. The changes in this PR are fully backwards compatible without the 2nd PR being merged/deployed yet.

More details are provided on the 2nd PR: https://github.com/near/near-discovery/pull/1103

### CODE REVIEW

File: src/ComponentSummary.jsx [hunk 1]
```diff
@@ -200,6 +201,34 @@ function normalizeMarkdown(text) {
   return text.trim();
 }
 
+function emitPinnedAppsGatewayEvent(isPinned) {
+  const app = {
+    authorAccountId: accountId,
+    displayName: metadata.name || widgetName,
+    componentName: widgetName,
+    imageUrl: metadata.image?.ipfs_cid
+      ? `https://i.near.social/large/https://ipfs.near.social/ipfs/${metadata.image.ipfs_cid}`
```
Comment (shelegdmitriy) [inline src/ComponentSummary.jsx:210]: https://i.near... and https://ipfs... is this the right image url?

File: src/ComponentSummary.jsx [hunk 2]
```diff
@@ -269,34 +299,105 @@ return (
...
-            <Button type="button" onClick={starOnClick} aria-label="Star this component">
-              {starIsActive ? (
-                <i className="bi bi-star-fill" style={{ color: "var(--amber10)" }} />
-              ) : (
-                <i className="bi bi-star" />
-              )}{" "}
-              {starCount}
-            </Button>
+            <Widget
+              src="near/widget/DIG.Tooltip"
```
Comment (shelegdmitriy) [inline src/ComponentSummary.jsx:303]:
```suggestion
              src="${REPL_ACCOUNT}/widget/DIG.Tooltip"
```

File: src/ComponentSummary.jsx [hunk 3]
```diff
@@ -269,34 +299,105 @@ return (
...
+            notifyAccountId: accountId,
+            onCommitStart: (pinIsActive) => {
+              emitPinnedAppsGatewayEvent(pinIsActive);
+            },
+            onCommitFailure: (pinIsActive) => {
+              emitPinnedAppsGatewayEvent(pinIsActive);
+            },
+            button: (pinCount, pinIsActive, pinOnClick) => (
+              <Widget
+                src="near/widget/DIG.Tooltip"
```
Comment (shelegdmitriy) [inline src/ComponentSummary.jsx:346]:
```suggestion
                src="${REPL_ACCOUNT}/widget/DIG.Tooltip"
```

File: src/ComponentSummary.jsx (hunk 2 above)
Comment (shelegdmitriy) [inline src/ComponentSummary.jsx:303]:
```suggestion
              src="${REPL_ACCOUNT}/widget/DIG.Tooltip"
```

File: src/ComponentSummary.jsx [hunk 4]
```diff
@@ -269,34 +299,105 @@ return (
...
-        >
-          {state.copiedShareUrl ? <i className="bi bi-16 bi-check"></i> : <i className="bi bi-16 bi-link-45deg"></i>}
-          Share
-        </Button>
-      </OverlayTrigger>
+        />
+      )}
+
+      <Widget
+        src="near/widget/DIG.Tooltip"
```
Comment (shelegdmitriy) [inline src/ComponentSummary.jsx:376]:
```suggestion
        src="${REPL_ACCOUNT}/widget/DIG.Tooltip"
```

**Review by shelegdmitriy — CHANGES_REQUESTED** [review]
Looks good! Left a few comments

File: src/ComponentSummary.jsx (hunk 3 above)
Comment (calebjacob) [reply src/ComponentSummary.jsx]: @shelegdmitriy Whoops! I had a brain fart here. Thanks so much for catching that! I did a global find and replace and found a few other spots that needed to use `${REPL_ACCOUNT}`.

File: src/ComponentSummary.jsx (hunk 1 above)
Comment (calebjacob) [reply src/ComponentSummary.jsx:210]: You're right, this URL does look weird. Let me double check what's going on here.

Comment (calebjacob) [reply src/ComponentSummary.jsx]: @shelegdmitriy I did some investigating and the URL was working (I had copied it from another place in our codebase). But I went ahead and pushed some changes to use the simpler `https://ipfs.near.social/ipfs/...` version. I can't event remember at this point where `https://i.near.social/large/https://ipfs.near.social...` originally came from.

**Review by shelegdmitriy — APPROVED** [review]

**Pull request merged**